### PR TITLE
[#9] [UI] [Android] As an anonymous user, I can see the login screen

### LINKED ISSUE #9: [UI] [Android] As an anonymous user, I can see the login screen
## Why

To provide the anonymous user with the identity verification method, show the login screen where the user can input his/her email and password, then access the main application's features.

## Acceptance criteria

After performing the splash screen, show the Nimble logo in the center, then the Nimble logo animates vertically to the top, and the login form fades in with the following:

- Two text fields for inputting credentials: one for email and one for password
- One log-in button

## Design

- [Design for the login screen](https://www.figma.com/proto/GjRPOjDyZ6f4EDL3wKarRK/Challenge---Mobile-App?node-id=31%3A195&scaling=scale-down)
- [Design for the animation from the splash screen to the login screen](https://www.figma.com/proto/GjRPOjDyZ6f4EDL3wKarRK/Challenge---Mobile-App?node-id=34%3A450&scaling=scale-down)

### PR DESCRIPTION
**Pull request by thiennguyen0196**
- Close #9 

## What happened 👀

- Create a general survey text field and button
- Create Sign In screen UI
- Create SurveyNavigation and SurveyDestination to navigate from the Splash screen to Sign In screen

## Insight 📝

- N/A

## Proof Of Work 📹

[Screen_recording_20230818_124629.webm](https://github.com/nimblehq/ic-kmm-kayla-bruce/assets/25218255/4daa2f00-a38b-4464-8ec5-c4c3d1b61740)

### CODE REVIEW

File: android/src/main/java/co/nimblehq/kaylabruce/kmmic/android/presentation/screen/signin/SignInScreen.kt [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    AnimatedVisibility(
+        visible = shouldShowLoginForm,
+        enter = fadeIn(animationSpec = tween(SHOW_LOGIN_DURATION))
+    ) {
+        Column(
+            verticalArrangement = Arrangement.spacedBy(20.dp),
+            modifier = Modifier
+                .fillMaxSize()
+                .wrapContentHeight()
+                .padding(horizontal = 24.dp)
```
Comment (nmint8m) [inline android/src/main/java/co/nimblehq/kaylabruce/kmmic/android/presentation/screen/signin/SignInScreen.kt:82]: You forget to add trailing comma `,`:

```suggestion
                .padding(horizontal = 24.dp),
```

File: android/src/main/java/co/nimblehq/kaylabruce/kmmic/android/presentation/screen/signin/SignInScreen.kt [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+                visualTransformation = PasswordVisualTransformation(),
+                imeAction = ImeAction.Done,
+                trailingIcon = {
+                    Text(
+                        text = "Forgot?",
+                        color = Color.White.copy(alpha = 0.5f),
+                        modifier = Modifier
+                            .padding(all = 16.dp)
+                    )
+                }
```
Comment (nmint8m) [inline android/src/main/java/co/nimblehq/kaylabruce/kmmic/android/presentation/screen/signin/SignInScreen.kt:104]: You forget to add trailing comma `,`:

```suggestion
                },
```

File: android/src/main/java/co/nimblehq/kaylabruce/kmmic/android/presentation/screen/signin/SignInScreen.kt [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+            )
+            SurveyButton(
+                modifier = Modifier
+                    .fillMaxWidth(),
+                text = "Login",
+                onClicked = {}
```
Comment (nmint8m) [inline android/src/main/java/co/nimblehq/kaylabruce/kmmic/android/presentation/screen/signin/SignInScreen.kt:110]: You forget to add trailing comma `,`:

```suggestion
                onClicked = {},
```

File: android/src/main/java/co/nimblehq/kaylabruce/kmmic/android/presentation/screen/signin/SignInScreen.kt [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+    Box(
+        contentAlignment = Alignment.Center,
+        modifier = Modifier.fillMaxSize()
+    ) {
+        Image(
+            painter = painterResource(id = R.drawable.im_background_blur),
+            contentDescription = null,
+            contentScale = ContentScale.Crop,
+            modifier = Modifier
```
Comment (nmint8m) [inline android/src/main/java/co/nimblehq/kaylabruce/kmmic/android/presentation/screen/signin/SignInScreen.kt:59]: You forget to add trailing comma `,`.

File: android/src/main/java/co/nimblehq/kaylabruce/kmmic/android/presentation/screen/signin/SignInScreen.kt [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+                .matchParentSize()
+        )
+
+        AnimatedVisibility(visible = true) {
+            Image(
+                painter = painterResource(id = R.drawable.ic_logo_white),
+                contentDescription = null,
+                modifier = Modifier
+                    .offset(animateLogoOffset.x.dp, animateLogoOffset.y.dp)
```
Comment (nmint8m) [inline android/src/main/java/co/nimblehq/kaylabruce/kmmic/android/presentation/screen/signin/SignInScreen.kt:68]: You forget to add trailing comma `,`.

File: android/src/main/java/co/nimblehq/kaylabruce/kmmic/android/presentation/screen/signin/SignInScreen.kt [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+            )
+        }
+    }
+
+    AnimatedVisibility(
+        visible = shouldShowLoginForm,
+        enter = fadeIn(animationSpec = tween(SHOW_LOGIN_DURATION))
```
Comment (nmint8m) [inline android/src/main/java/co/nimblehq/kaylabruce/kmmic/android/presentation/screen/signin/SignInScreen.kt:75]: You forget to add trailing comma `,`.

File: android/src/main/java/co/nimblehq/kaylabruce/kmmic/android/presentation/screen/signin/SignInScreen.kt [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+    }
+
+    LaunchedEffect(Unit) {
+        delay(SHOW_LOGIN_DURATION.toLong())
+        shouldShowLoginForm = true
+    }
+
+    Box(
+        contentAlignment = Alignment.Center,
+        modifier = Modifier.fillMaxSize()
```
Comment (nmint8m) [inline android/src/main/java/co/nimblehq/kaylabruce/kmmic/android/presentation/screen/signin/SignInScreen.kt:53]: You forget to add trailing comma `,`.

File: android/src/main/java/co/nimblehq/kaylabruce/kmmic/android/presentation/screen/signin/SignInScreen.kt [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+@Composable
+fun SignInScreen() {
+    var email by remember { mutableStateOf("") }
+    var password by remember { mutableStateOf("") }
+    var logoOffset by remember { mutableStateOf(initialLogoOffset) }
+    var shouldShowLoginForm by remember { mutableStateOf(false) }
+    val animateLogoOffset by animateOffsetAsState(
+        targetValue = logoOffset,
+        animationSpec = tween(durationMillis = SHOW_LOGIN_DURATION),
+        label = ""
```
Comment (nmint8m) [inline android/src/main/java/co/nimblehq/kaylabruce/kmmic/android/presentation/screen/signin/SignInScreen.kt:39]: You forget to add trailing comma `,`.

File: android/src/main/java/co/nimblehq/kaylabruce/kmmic/android/presentation/screen/common/SurveyButton.kt [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+
+@Composable
+fun SurveyButton(
+    modifier: Modifier = Modifier,
+    text: String,
+    onClicked: () -> Unit,
```
Comment (nmint8m) [inline android/src/main/java/co/nimblehq/kaylabruce/kmmic/android/presentation/screen/common/SurveyButton.kt:17]: Should it be `onClick`? I think we should name this one similarly to `Button`'s `onClick`. What do you think?

File: android/src/main/java/co/nimblehq/kaylabruce/kmmic/android/presentation/screen/common/SurveyTextField.kt [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.text.input.*
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+
+@Composable
+fun SurveyTextField(
+    modifier: Modifier = Modifier,
+    text: String,
+    onTextChanged: (String) -> Unit,
```
Comment (nmint8m) [inline android/src/main/java/co/nimblehq/kaylabruce/kmmic/android/presentation/screen/common/SurveyTextField.kt:24]: How about `onTextChange`?

File: android/src/main/java/co/nimblehq/kaylabruce/kmmic/android/presentation/screen/common/SurveyTextField.kt [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+            keyboardType = keyboardType,
+            imeAction = imeAction
+        ),
+        keyboardActions = KeyboardActions(onDone = { focusManager.clearFocus() }),
+        trailingIcon = trailingIcon,
+        modifier = modifier
+            .fillMaxWidth()
+            .heightIn(min = 56.dp)
+            .clip(RoundedCornerShape(8.dp))
+            .semantics { this.contentDescription = contentDescription }
```
Comment (nmint8m) [inline android/src/main/java/co/nimblehq/kaylabruce/kmmic/android/presentation/screen/common/SurveyTextField.kt:65]: Please add trailing comma.

File: android/src/main/java/co/nimblehq/kaylabruce/kmmic/android/presentation/screen/common/SurveyTextField.kt [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+            focusedIndicatorColor = Color.Transparent,
+            unfocusedIndicatorColor = Color.Transparent,
+            disabledIndicatorColor = Color.Transparent,
+            errorIndicatorColor = Color.Transparent,
+            placeholderColor = Color(0x4DFFFFFF)
+        ),
+        visualTransformation = visualTransformation,
+        keyboardOptions = KeyboardOptions(
+            keyboardType = keyboardType,
+            imeAction = imeAction
```
Comment (nmint8m) [inline android/src/main/java/co/nimblehq/kaylabruce/kmmic/android/presentation/screen/common/SurveyTextField.kt:57]: Please add trailing comma.

File: android/src/main/java/co/nimblehq/kaylabruce/kmmic/android/presentation/screen/common/SurveyTextField.kt [hunk 13]
```diff
@@ -0,0 +1,77 @@
...
+        singleLine = singleLine,
+        colors = TextFieldDefaults.textFieldColors(
+            textColor = Color.White,
+            cursorColor = Color.White,
+            backgroundColor = Color(0x2EFFFFFF),
+            focusedIndicatorColor = Color.Transparent,
+            unfocusedIndicatorColor = Color.Transparent,
+            disabledIndicatorColor = Color.Transparent,
+            errorIndicatorColor = Color.Transparent,
+            placeholderColor = Color(0x4DFFFFFF)
```
Comment (nmint8m) [inline android/src/main/java/co/nimblehq/kaylabruce/kmmic/android/presentation/screen/common/SurveyTextField.kt:52]: Please add trailing comma.

File: android/src/main/java/co/nimblehq/kaylabruce/kmmic/android/presentation/screen/common/SurveyTextField.kt [hunk 14]
```diff
@@ -0,0 +1,77 @@
...
+    trailingIcon: @Composable (() -> Unit)? = null,
+) {
+    val focusManager = LocalFocusManager.current
+    TextField(
+        value = text,
+        onValueChange = onTextChanged,
+        placeholder = {
+            Text(
+                text = placeholder,
+                style = MaterialTheme.typography.body1
```
Comment (nmint8m) [inline android/src/main/java/co/nimblehq/kaylabruce/kmmic/android/presentation/screen/common/SurveyTextField.kt:40]: Please add trailing comma.

File: android/src/main/java/co/nimblehq/kaylabruce/kmmic/android/presentation/screen/signin/SignInScreen.kt [hunk 15]
```diff
@@ -0,0 +1,120 @@
...
+                onTextChanged = { password = it },
+                placeholder = "Password",
+                visualTransformation = PasswordVisualTransformation(),
+                imeAction = ImeAction.Done,
+                trailingIcon = {
+                    Text(
+                        text = "Forgot?",
+                        color = Color.White.copy(alpha = 0.5f),
+                        modifier = Modifier
+                            .padding(all = 16.dp)
```
Comment (nmint8m) [inline android/src/main/java/co/nimblehq/kaylabruce/kmmic/android/presentation/screen/signin/SignInScreen.kt:102]: Please add trailing comma.

**Review by nmint8m — COMMENTED** [review]
According to this [Kotlin convention](https://kotlinlang.org/docs/coding-conventions.html#trailing-commas) we should include trailing comma for code. I'm not sure how we can archive this easier. Do we have any linter for this @thiennguyen0196 ?

**Comment by thiennguyen0196** [comment]
@nmint8m I have resolved your concerns, unfortunately, we have not had any linter for the trailing commas yet, but let me do some research about it and will implement it later if possible 👍 Nice catch btw

**Review by nmint8m — APPROVED** [review]
Look good to me

**Review by markgravity — APPROVED** [review]

**Pull request merged**